smalltalksrc/VMMakerTests/VMSpurOldSpaceGarbageCollectorTest.class.st

```diff
@@ -220,6 +220,61 @@ VMSpurOldSpaceGarbageCollectorTest >> testAnOldObjectReferencedFromVMVariableSho
 	self assertHashOf: self keptObjectInVMVariable1 equals: hash
 ]
 
+{ #category : #ephemerons }
+VMSpurOldSpaceGarbageCollectorTest >> testCompactEphemeronQueuePass1 [
+
+	| ephemeron |
+
+	"For first compaction pass"
+	"Hole, then mourn queue"
+	self newOldSpaceObjectWithSlots: 0.
+	memory initializeMournQueue.
+
+	"For second compaction pass"
+	"Hole, then ephemeron. Does not really need to be an ephemeron for this tests, any object does it"
+	self newOldSpaceObjectWithSlots: 0.	
+	ephemeron := self newOldSpaceObjectWithSlots: 0.
+	self keepObjectInVMVariable1: ephemeron.
+	memory push: ephemeron onObjStack: memory mournQueue.
+
+	"Compact"
+	memory garbageCollectForSnapshot.
+	
+	"Assert?"
+	self assert: (memory sizeOfObjStack: memory mournQueue) equals: 1.
+	self assert: memory dequeueMourner equals: self keptObjectInVMVariable1.
+]
```

Comment:
In this test, there are 2 compaction passes or only one? If there is only one we need to fix the comments (and maybe the test's name)

Reply:
```suggestion
VMSpurOldSpaceGarbageCollectorTest >> testCompactEphemeronQueuePass1 [

	| ephemeron |

	"Hole, then mourn queue"
	self newOldSpaceObjectWithSlots: 0.
	memory initializeMournQueue.

	"Hole, then ephemeron. Does not really need to be an ephemeron for this tests, any object does it"
	self newOldSpaceObjectWithSlots: 0.	
	ephemeron := self newOldSpaceObjectWithSlots: 0.
	self keepObjectInVMVariable1: ephemeron.
	memory push: ephemeron onObjStack: memory mournQueue.

	"Compact in one pass"
	memory garbageCollectForSnapshot.
	
	self assert: (memory sizeOfObjStack: memory mournQueue) equals: 1.
	self assert: memory dequeueMourner equals: self keptObjectInVMVariable1.
]
```